### PR TITLE
[#135] Add documentation about internal module structure

### LINKED ISSUE #135: Add internal documentation regarding module structure
> current structure is very mysterious to someone who did not create it.

So we should write documentation where to lookup if you want to find specific function.

### PR DESCRIPTION
**Pull request by ChShersh**
Resolves #135

### CODE REVIEW

File: src/Universum.hs [hunk 1]
```diff
@@ -1,14 +1,61 @@
...
+
+This documentation section contains description of internal module structure to
+help navigate between modules, search for interested functionality and
+understand where you need to put your new changes.
+
+Functions and types are distributed across multiple modules and grouped by
+meaning or __theme__. Name of the module should give you hints regarding what
+this module contains. Some /themes/ contains a lot of reexported stuff and a lot
+of our own. To make it easier to maintain and understand such huge chunks of
+functions, all reexported staff is moved into separate module with name
```
Comment (NaeosPsy) [inline src/Universum.hs]: staff -> stuff

File: src/Universum.hs [hunk 2]
```diff
@@ -1,14 +1,61 @@
...
+module to replace default 'Prelude' with better one.
+
+@
+\{\-\# LANGUAGE NoImplicitPrelude \#\-\}
+
+__import__ "Universum"
+@
+
+This documentation section contains description of internal module structure to
+help navigate between modules, search for interested functionality and
```
Comment (NaeosPsy) [inline src/Universum.hs]: interested functionality -> interesting functionality?

File: src/Universum.hs [hunk 3]
```diff
@@ -1,14 +1,61 @@
...
+understand where you need to put your new changes.
+
+Functions and types are distributed across multiple modules and grouped by
+meaning or __theme__. Name of the module should give you hints regarding what
+this module contains. Some /themes/ contains a lot of reexported stuff and a lot
```
Comment (NaeosPsy) [inline src/Universum.hs]: "Some /themes/ contain a great amount of both reexported functions and functions of our own. To make it easier to understand these huge chunks of functions, ... "

File: README.md [hunk 4]
```diff
@@ -122,6 +125,10 @@ If you're using [Emacs](https://www.gnu.org/software/emacs/), you can
 [modify your configs](https://github.com/serokell/universum/issues/8#issuecomment-276444879)
 a little bit if you don't want to type `import Universum` manually every time.
 
+If you want to get familiar with `universum` internal structure you can just
```
Comment (NaeosPsy) [inline README.md]: "," after structure

File: README.md [hunk 5]
```diff
@@ -122,6 +125,10 @@ If you're using [Emacs](https://www.gnu.org/software/emacs/), you can
 [modify your configs](https://github.com/serokell/universum/issues/8#issuecomment-276444879)
```
Comment (NaeosPsy) [inline README.md]: If you're using [Emacs](https://www.gnu.org/software/emacs/) and don't want to type `import Universum` manually every time, you can [modify your configs](https://github.com/serokell/universum/issues/8#issuecomment-276444879) a little bit.

File: README.md [hunk 6]
```diff
@@ -12,10 +12,13 @@ Universum
 
 1. **Excellent documentation**: tutorial, migration guide from `Prelude`,
    Haddock with examples for (almost) every function,
-   all examples are tested with [`doctest`](http://hackage.haskell.org/package/doctest).
+   all examples are tested with [`doctest`](http://hackage.haskell.org/package/doctest),
+   documenation regarding internal module structure.
 2. `universum`-specific [HLint](http://hackage.haskell.org/package/hlint) rules:
    [`.hlint.yaml`](https://github.com/serokell/universum/blob/master/.hlint.yaml)
-3. Focus on safety, convenience and efficiency.
+3. Very few LiquidHaskell properties. At least LiquidHaskell is on Travis CI and
```
Comment (NaeosPsy) [inline README.md]: "Only a few LiquidHaskell properties right now, but LiquidHaskell is on Travis CI and other properties are just waiting to be added!" ?

File: src/Universum.hs [hunk 7]
```diff
@@ -1,14 +1,61 @@
...
+of our own. To make it easier to maintain and understand such huge chunks of
+functions, all reexported staff is moved into separate module with name
+@Universum.SomeTheme.Reexport@ and our own functions and types are in
+@Universum.SomeTheme.SomeName@. For example, see modules
+"Universum.Container.Class" and "Universum.Container.Reexport".
```
[the remark this answers is not in the document]
Comment (ChShersh) [reply src/Universum.hs]: Probably looks better... But it's too tedious to format every module like that... And we don't have best practices for formatting Haddock..

**Pull request merged**